forge/src/main/java/me/lucko/luckperms/forge/ForgeSchedulerAdapter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of LuckPerms, licensed under the MIT License.
+ *
+ *  Copyright (c) lucko (Luck) <luck@lucko.me>
+ *  Copyright (c) contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+package me.lucko.luckperms.forge;
+
+import me.lucko.luckperms.common.plugin.scheduler.AbstractJavaScheduler;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinWorkerThread;
+
+public class ForgeSchedulerAdapter extends AbstractJavaScheduler {
+    private final Executor sync;
+
+    public ForgeSchedulerAdapter(LPForgeBootstrap bootstrap) {
+        super(new WorkerThreadFactory());
+        this.sync = r -> bootstrap.getServer().orElseThrow(() -> new IllegalStateException("Server not ready")).executeBlocking(r);
+    }
+
+    @Override
+    public Executor sync() {
+        return this.sync;
+    }
+
+    private static final class WorkerThreadFactory implements ForkJoinPool.ForkJoinWorkerThreadFactory {
+
+        @Override
+        public ForkJoinWorkerThread newThread(ForkJoinPool pool) {
+            return new WorkerThread(pool);
+        }
+    }
+
+    private static final class WorkerThread extends ForkJoinWorkerThread {
+
+        private WorkerThread(ForkJoinPool pool) {
+            super(pool);
+            setContextClassLoader(Thread.currentThread().getContextClassLoader());
```

Comment:
Why is this necessary?

Some comments would be good here.

Reply:
Comments added, I originally documented the reason for this in the commit message https://github.com/LuckPerms/LuckPerms/pull/3262/commits/ec8b8565977a22d42883278968f1f84bed3e93a9